src/ui/common/src/components/pages/artifact/id/index.tsx

```diff
@@ -0,0 +1,136 @@
+} from '../../../../reducers/workflow';
+import { AppDispatch, RootState } from '../../../../stores/store';
+import UserProfile from '../../../../utils/auth';
+import DefaultLayout from '../../../layouts/default';
+import KeyValueTable from '../../../tables/KeyValueTable';
+import StickyHeaderTable from '../../../tables/StickyHeaderTable';
+import { LayoutProps } from '../../types';
+
+type ArtifactDetailsHeaderProps = {
+    artifactName: string;
+    createdAt?: string;
+    sourceLocation?: string;
+};
+
+const ArtifactDetailsHeader: React.FC<ArtifactDetailsHeaderProps> = ({
+    artifactName,
+    // TODO: add these back once we have support for getting createdAt and sourceLocation.
+    //createdAt,
+    //sourceLocation,
+}) => {
+    return (
+        <Box width="100%" display="flex" alignItems="center">
+            <FontAwesomeIcon height="24px" width="24px" style={{ marginRight: '8px' }} icon={faCircleCheck} color={'green'} />
+            <Typography variant="h4" component="div">
+                {artifactName}
+            </Typography>
+            {/* <Typography marginTop="4px" variant="caption" component="div">
+                Created: {createdAt}
+            </Typography>
+            <Typography variant="caption" component="div">
+                Source: <Link>{sourceLocation}</Link>
+            </Typography> */}
+        </Box>
+    );
+};
+
+type ArtifactDetailsPageProps = {
+    user: UserProfile;
+    Layout?: React.FC<LayoutProps>;
+};
+
+const ArtifactDetailsPage: React.FC<ArtifactDetailsPageProps> = ({
+    user,
+    Layout = DefaultLayout,
+}) => {
+    const dispatch: AppDispatch = useDispatch();
+    const { workflowDagResultId, artifactId } = useParams();
+    const artifactResult: ArtifactResult | null = useSelector(
+        (state: RootState) => {
+            // First, check if there are any keys in the artifactResults object.
+            const artifactResults = state.workflowReducer.artifactResults;
+            if (Object.keys(artifactResults).length < 1) {
+                return null;
+            }
+
+            return artifactResults[artifactId];
+        }
+    );
+
+    useEffect(() => {
+        document.title = 'Artifact | Aqueduct';
+
+        // Check and see if we are loading the artifact result
+        if (!artifactResult) {
+            dispatch(
+                handleGetArtifactResults({
+                    apiKey: user.apiKey,
+                    workflowDagResultId,
+                    artifactId,
+                })
+            );
+        }
+    }, []);
+
+    if (!artifactResult || !artifactResult.result) {
+        return (
+            <Layout user={user}>
+                <CircularProgress />
+            </Layout>
+        );
+    }
+
+    const parsedData = JSON.parse(artifactResult.result.data);
+
+    return (
+        <Layout user={user}>
+            <Box width={'800px'}>
+                <Box width="100%">
+                    <Box width="100%" display="flex">
```

Comment:
Going to take this header and make into more of a common component in the future. This header is used across our new mocks.